src/UI/Components/Diagram/Canvas.tsx

```diff
@@ -54,6 +55,21 @@ const Canvas = ({
     service: mainServiceName,
   });
 
+  const handleLooseEmbeddedEvent = (event) => {
+    const customEvent = event as CustomEvent;
+    const eventData: { kind: "remove" | "add"; id: string } = JSON.parse(
+      customEvent.detail,
+    );
+    if (eventData.kind === "remove") {
+      setLooseEmbedded((prevSet) => {
+        prevSet.delete(eventData.id);
+        return new Set(prevSet);
+      });
+    } else {
+      setLooseEmbedded((prevSet) => new Set(prevSet.add(eventData.id)));
```

Comment:
same here as above. It's always best to prefer immutability principles to prevent side effects on the long run. Mutating the data should be done deliberately and for specific reasons, which is not the case here I believe.